### DISCUSSION TITLE
Build IPA for Native iOS project Failed

### DISCUSSION THREAD

**Discussion by neerajpal2407** [codemagic-ci-cd/codemagic-docs#2546] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

When Creating build for Native iOS by codeMagic then it gives error:

usage: xcode-project build-ipa [-h] [--log-stream {stderr,stdout}] [--no-color] [--version] [-s] [-v] [--project XCODE_PROJECT_PATH] [--workspace XCODE_WORKSPACE_PATH] [--target TARGET_NAME] [--config CONFIGURATION_NAME] [--scheme SCHEME_NAME] [--clean] [--no-show-build-settings] [--archive-directory ARCHIVE_DIRECTORY] [--archive-flags ARCHIVE_FLAGS] [--archive-xcargs ARCHIVE_XCARGS] [--ipa-directory IPA_DIRECTORY] [--export-options-plist EXPORT_OPTIONS_PLIST] [--export-flags EXPORT_FLAGS] [--export-xcargs EXPORT_XCARGS] [--remove-xcarchive] [--disable-xcpretty] [--xcpretty-options XCPRETTY_OPTIONS]
xcode-project build-ipa: error: argument --workspace: Path "/Users/builder/clone/demoAppForCICD.xcworkspace" does not exist
<img width="740" alt="Screenshot 2024-05-21 at 1 09 22 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/119958959/f5964d92-05b3-4ff2-bdf3-38fa021977a7">
this is my project folder
<img width="744" alt="Screenshot 2024-05-21 at 4 16 00 PM" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/119958959/e6d0088c-1ba4-4ee2-a074-1aa32057c1dd">
This is my Yaml File 

Build failed :|
Step 4 script `Build ipa for distribution` exited with status code 2

### Expected results

Please assist me why this issue occure

### Actual results

Please assist me why this issue occure

**Comment by dtrdic** [comment]
Hi @neerajpal2407 the error message suggests that your `.xcworkspace` is on a different location than the one you have specified 
`--workspace: Path "/Users/builder/clone/demoAppForCICD.xcworkspace" does not exist`
maybe is on this location `--workspace "$CM_BUILD_DIR/demoAppForCICD/$XCODE_WORKSPACE" `

But the safest way is to double-check it by yourself on the build machine.

1. Click on enable SSH/VNC checkbox
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/034f392f-5dd5-452a-bdf7-639775991e1c)

2. the Explore build machine via SSH or VNC client button will appear
![image](https://github.com/codemagic-ci-cd/codemagic-docs/assets/38918425/2ffb51dd-721e-44f9-8dd3-19c7fd305815)

3.Then just enter the host address, username, and password into VNC viewer on your local machine.
﻿The source code of your app is cloned to a folder called `clone`, check in there where actually the `.xcworkspace` is located

**Comment by neerajpal2407** [comment]
Sorry but i would like to inform you that from now we are not going to use
CodeMagic for company So don't need to reply us it's happened due to late
reply and low efficient of developer to problem solve
